In [1]:
%load_ext sql

In [2]:
%sql mysql://admin:sql_2020@lmu-sql.crkxqhoxro46.us-east-2.rds.amazonaws.com/sql_project

'Connected: admin@sql_project'

Question: Create a table that displays who the top performing artists are?

Business Justification: By seeing the artists who have the most songs in the top 50 chart, we can see which artist creates the most top tracks. We will pursue that artist to try to create deals and try to tailor some of the Apple Music marketing with the use of those better performing artists. 

SQL features: Common Table Expression, Join, Group by

In [9]:
%%sql
#CTE to see which artists have the most top songs
WITH artist_top_song_count AS (
    SELECT artist_name, track_name
    FROM top_50_songs 
    JOIN artists 
        ON artists.artist_id = top_50_songs.artist_id
)

SELECT artist_name, COUNT(*) AS num_top_songs
FROM artist_top_song_count
GROUP BY artist_name
ORDER BY num_top_songs DESC;

 * mysql://admin:***@lmu-sql.crkxqhoxro46.us-east-2.rds.amazonaws.com/sql_project
38 rows affected.


[('Bad Bunny', 5),
 ('Dua Lipa', 3),
 ('The Weeknd', 2),
 ('Doja Cat', 2),
 ('DaBaby', 2),
 ('Lewis Capaldi', 2),
 ('Billie Eilish', 2),
 ('J Balvin', 2),
 ('Tones And I', 1),
 ('Surfaces', 1),
 ('Sam Smith', 1),
 ('Surf Mesa', 1),
 ('MEDUZA', 1),
 ('Roddy Ricch', 1),
 ('Jack Harlow', 1),
 ('Shawn Mendes', 1),
 ('Lil Mosey', 1),
 ('Megan Thee Stallion', 1),
 ('Camilo', 1),
 ('DripReport', 1),
 ('KAROL G', 1),
 ('Harry Styles', 1),
 ('Eminem', 1),
 ('Drake', 1),
 ('Arizona Zervas', 1),
 ('Maroon 5', 1),
 ('JP Saxe', 1),
 ('SAINt JHN', 1),
 ('Justin Bieber', 1),
 ('Future', 1),
 ('Rauw Alejandro', 1),
 ('BENEE', 1),
 ('Post Malone', 1),
 ('Regard', 1),
 ('Powfu', 1),
 ('Trevor Daniel', 1),
 ('Playboi Carti', 1),
 ('Twenty One Pilots', 1)]

Reccomendaion:

Based on these results we see that the artist that performs the most top songs is Bad Bunny. As for my reccomendation, I believe that Apple Music should strike some sort of deal with Bad Bunny, for example, have him release his music exclusively on Apple Music for a week before releasing on other platforms. Additionally, if Bad Bunny would not agree to that, Dua Lipa,  who follows Bad Bunny's 5 top songs with 3 top songs, could be another target for a similar promotional plan.



Question: Create a query that displays the the top genres and the percentage of each age group that listens to that genre and the average artist popularity for that specific genre.

Business Justification: By showing the most popular genres and their respective age groups, we can choose which age group to market towards and form a target market for Apple Music using that. Additionally, marketing can be ramped up for specific genres in specific age groups while other genres and age groups not so much.

SQL features: View, Join, Group By


In [43]:
%%sql
CREATE OR REPLACE VIEW top_songs_by_artists_with_genre AS
    SELECT artists.artist_popularity, genres.genre_id, genres.genre_name, genres.age_16_19, 
        genres.age_20_24, genres.age_25_34, genres.age_35_44, genres.age_45_54, genres.age_55_64, genres.age_65_plus
    FROM top_50_songs
    JOIN artists
        ON top_50_songs.artist_id = artists.artist_id
    JOIN genres
        ON artists.genre_id = genres.genre_id;

SELECT 
    genre_id, genre_name, age_16_19, 
    age_20_24, age_25_34, 
    age_35_44, age_45_54, 
    age_55_64, age_65_plus, 
    AVG(artist_popularity) as avg_artist_popularity_per_genre, 
    COUNT(*) AS num_of_top_songs_in_genre
FROM top_songs_by_artists_with_genre
GROUP BY genre_id
ORDER BY COUNT(*) DESC, AVG(artist_popularity) DESC;



 * mysql://admin:***@lmu-sql.crkxqhoxro46.us-east-2.rds.amazonaws.com/sql_project
0 rows affected.
5 rows affected.


[(1, 'pop', 52, 54, 56, 46, 42, 25, 19, Decimal('90.2857'), 21),
 (3, 'rap', 48, 54, 42, 33, 15, 5, 1, Decimal('88.3571'), 14),
 (8, 'latin', 30, 35, 36, 34, 21, 13, 5, Decimal('95.5000'), 10),
 (5, 'edm', 20, 33, 15, 21, 9, 5, 1, Decimal('78.0000'), 3),
 (2, 'rock', 26, 36, 44, 46, 62, 60, 48, Decimal('87.5000'), 2)]

Reccomendation: 

We see by far that pop has the most songs in the top 50 charts. Additionally pop has an average artist popularity of 90.2857 out of 100. Therefore if resources were to be given towards marketing the most popular genre it would be pop. Additionally, the age groups demonstrate that more than half of those from age 16-34 listen to pop music. Therefore, marketing should be done towards Millenials and Gen Z.

